realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -2217,19 +2217,19 @@ private void checkQueryIsNotReused() {
     }
 
     private long getSourceRowIndexForFirstObject() {
-        long rowIndex = this.query.find();
-        if (rowIndex < 0) {
-            return rowIndex;
-        }
-        if (this.view != null) {
-            return view.getTargetRowIndex(rowIndex);
-        } else if (table instanceof TableView){
-            return ((TableView) table).getSourceRowIndex(rowIndex);
-        } else {
-            return rowIndex;
-        }
+        long tableRowIndex = this.query.find();
+        return tableRowIndex;
+//        if (tableRowIndex < 0) {
```

Comment:
No, they can go. From what I can tell the implementation is correct, but I was on a goose chase because of the bug in core revealed by one of the unit tests.